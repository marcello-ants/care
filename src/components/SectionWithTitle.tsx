import { makeStyles } from '@material-ui/core';
import { Typography } from '@care/react-component-lib';

interface SectionWithTitleProps {
  title: string;
  learnMoreLink?: string;
  children: string;
}

const useStyles = makeStyles((theme) => ({
  title: {
    marginBottom: theme.spacing(1),
  },
  body: {
    marginTop: theme.spacing(1),
    marginBottom: theme.spacing(4),
    fontSize: '14px',
  },
}));

function SectionWithTitle({ title, learnMoreLink, children }: SectionWithTitleProps) {
  const classes = useStyles();

  return (
    <div>
      <Typography variant="h2" className={classes.title}>
        {title}
      </Typography>
      <Typography variant="body1" className={classes.body}>
        {children} {learnMoreLink ? <a href={learnMoreLink}>Learn more</a> : null}
      </Typography>
    </div>
  );
}

SectionWithTitle.defaultProps = {
  learnMoreLink: undefined,
};

export default SectionWithTitle;
