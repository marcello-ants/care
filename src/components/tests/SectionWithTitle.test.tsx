import React from 'react';
import { render } from '@testing-library/react';
import { MockedProvider, MockedResponse } from '@apollo/client/testing';
// Internal Dependencies
import SectionWithTitle from '../SectionWithTitle';

const props = {
  title: 'How do you take advantage of these tax credits',
  learnMoreLink: 'https://www.care.com/homepay',
  children:
    'In order to take advantage of these tax credits, you must pay your nanny on the books. Care.com Homepay can help.',
};

const mocks: MockedResponse[] = [];

afterEach(() => {
  jest.clearAllMocks();
});

describe('SectionWithTitle - test', () => {
  it('matches snapshot', async () => {
    const { asFragment } = render(
      <MockedProvider mocks={mocks} addTypename>
        <SectionWithTitle {...props} />
      </MockedProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
