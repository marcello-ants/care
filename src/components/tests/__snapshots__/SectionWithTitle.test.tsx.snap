// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SectionWithTitle - test matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <h2
      class="MuiTypography-root makeStyles-title-1 MuiTypography-h2"
    >
      How do you take advantage of these tax credits
    </h2>
    <p
      class="MuiTypography-root makeStyles-body-2 MuiTypography-body1"
    >
      In order to take advantage of these tax credits, you must pay your nanny on the books. Care.com Homepay can help. 
      <a
        href="https://www.care.com/homepay"
      >
        Learn more
      </a>
    </p>
  </div>
</DocumentFragment>
`;
